app/lib/identity_tijuana/member_sync.rb

```diff
@@ -6,25 +6,55 @@ def self.export_member(member, sync_id)
       begin
         return if member.ghosting_started?
 
-        ext_id = MemberExternalId.find_by(system: 'tijuana', member: member)
-        user_id = ext_id.external_id if ext_id.present?
-        if user_id.present?
-          user = User.find_by(id: user_id)
-          ext_id.destroy! if user.blank?
+        ext_ids = MemberExternalId.where(system: 'tijuana', member: member).to_a
+        # Destroy any dangling external ID references, remove any
+        # external ids from the list if not found
+        ext_ids.select do |ext_id|
+          user = User.find_by(id: ext_id.external_id.to_i)
+          if user.blank?
+            Rails.logger.info(
+              "[IdentityTijuana::export_member]: " \
+              "Removing dangling external id for Id member #{member.id}, " \
+              "TJ user #{ext_id.external_id} " \
+              "(sync_id=#{sync_id}, sync_type=#{sync_type}, " \
+              "sync_direction=export_member)"
+            )
+            ext_id.destroy!
+          end
+          user.present?
```

Comment:
dangling eval

Reply:
Actually no, that's the return val for the block for the `ext_ids.select` call. (yet another pet Ruby hate)